### PR TITLE
Add support for hf datasets summarization

### PR DESCRIPTION
**Pull request by Prikshit7766**
This PR aims on Adding Support of HF dataset for `summarization` task.

## Checklist:

- [ ] I've added Google style docstrings to my code.
- [ ] I've used `pydantic` for typing when/where necessary.
- [ ] I have linted my code
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: langtest/utils/custom_types/sample.py [hunk 1]
```diff
@@ -637,14 +638,29 @@ def run(self, model, **kwargs):
...
+            dataset_name = self.dataset_name.split('-')[0].lower()
+           
+            prompt_template = kwargs.get(
+                'user_prompt', default_user_prompt.get(dataset_name, ""))
+            
+         
+            self.expected_results = model(text={'context': self.original},
+                                        prompt={"template": prompt_template, 'input_variables': ["context"]})
+            self.actual_results = model(text={'context': self.test_case},
+                                        prompt={"template": prompt_template, 'input_variables': ["context"]})
```
Comment (JulesBelveze) [inline langtest/utils/custom_types/sample.py:651]: This snippet is shared in both conditions can you please factorise it?

File: langtest/datahandler/datasource.py [hunk 2]
```diff
@@ -610,8 +610,21 @@ def __init__(self, dataset_name: str):
...
+        """
+        Check if the 'datasets' package is installed and import the load_dataset function.
+        Raises an error if the package is not found.
+        """
+        try:
+            from datasets import load_dataset
+        except ImportError:
+            raise ModuleNotFoundError(
+                "The 'datasets' package is not installed. Please install it using 'pip install datasets'.")
+        self.load_dataset = load_dataset
```
Comment (JulesBelveze) [inline langtest/datahandler/datasource.py:625]: Please use the `langtest.utils.lib_manager.try_import_lib` function instead

**Review by JulesBelveze — CHANGES_REQUESTED** [review]
Few comments:
1. It would be better to pass a parameter `task` when instantiating the `HuggingFaceDataset`. This will go in hand with the others `*Dataset`.
2. It would be great to have a `HuggingFaceData.load_data` method that will then redirect to the loading method according to the task. That way we will always call the same function from the `Harness`.
3. Please add `SummarizationSample` to the TypeVar `Sample`

**Comment by RakshitKhajuria** [comment]
> Few comments:
> 
> 1. It would be better to pass a parameter `task` when instantiating the `HuggingFaceDataset`. This will go in hand with the others `*Dataset`.
> 2. It would be great to have a `HuggingFaceData.load_data` method that will then redirect to the loading method according to the task. That way we will always call the same function from the `Harness`.
> 3. Please add `SummarizationSample` to the TypeVar `Sample`

Thanks for the review @JulesBelveze :)

**Review by JulesBelveze — APPROVED** [review]
LGTM

**Pull request merged**